extmod/uasyncio/stream.py

```diff
@@ -0,0 +1,141 @@
+# MicroPython uasyncio module
+# MIT license; Copyright (c) 2019-2020 Damien P. George
+
+from . import core
+
+
+class Stream:
+    def __init__(self, s, e={}):
+        self.s = s
+        self.e = e
+        self.out_buf = b""
+
+    def get_extra_info(self, v):
+        return self.e[v]
+
+    async def __aenter__(self):
+        return self
+
+    async def __aexit__(self, exc_type, exc, tb):
+        await self.close()
+
+    def close(self):
+        pass
```

Comment:
Why does this not initiate the close? It means that one has to await wait_closed for anything to happen, which is not necessarily desirable. CPython does seem to initiate the close: https://github.com/python/cpython/blob/7ddcd0caa4c2e6b43265df144f59c5aa508a94f2/Lib/asyncio/streams.py#L352-L353

Reply:
I never got to the bottom of how close behaves in a non-blocking case, so this is still a TODO.  But it's definitely important to call `wait_closed()` for the socket to be properly closed.